lib/ASRouter.jsm

```diff
@@ -1777,6 +1775,62 @@ class _ASRouter {
     this.onMessage({ data: action, target });
   }
 
+  async sendNewTabMessage(target, options = {}) {
+    const { endpoint } = options;
+    let message;
+
+    // Load preview endpoint for snippets if one is sent
+    if (endpoint) {
+      await this._addPreviewEndpoint(endpoint.url, target.portID);
+    }
+
+    // Load all messages
+    await this.loadMessagesFromAllProviders();
+
+    if (endpoint) {
+      message = await this.handleMessageRequest({ provider: "preview" });
+    } else {
+      // On new tab, send cards if they match; othwerise send a snippet
+      message =
+        (await this.handleMessageRequest({
+          provider: "onboarding",
+          template: "extended_triplets",
+        })) || (await this.handleMessageRequest({ provider: "snippets" }));
+    }
+
+    if (endpoint && message) {
```

Comment:
Maybe this is a little more clear if moved after line 1791 inside the 
````javascript
if (endpoint) {
  message =
````

block.

Reply:
Same for the `non-preview-message` setState branch.